### PR TITLE
EPMRPP-57209 || Launch event publishing impl

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java [hunk 1]
```diff
@@ -68,22 +76,26 @@ public StartLaunchRS startLaunch(ReportPortalUser user, ReportPortalUser.Project
...
 
-		Launch launch = new LaunchBuilder().addStartRQ(request)
-				.addAttributes(request.getAttributes())
-				.addProject(projectDetails.getProjectId())
-				.addUserId(userId)
-				.get();
-		launchRepository.save(launch);
-		launchRepository.refresh(launch);
-		messageBus.publishActivity(new LaunchStartedEvent(TO_ACTIVITY_RESOURCE.apply(launch), user.getUserId(), user.getUsername()));
+		eventPublisher.publishEvent(new LaunchEvent(savedLaunch.getId(), START_KEY));
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java:93]: Can it be a separate class for this type of event?

File: src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java [hunk 2]
```diff
@@ -65,6 +71,7 @@ public void handleEvent(LaunchFinishedEvent launchFinishedEvent, Project project
 		} else {
 			logIndexer.indexLaunchLogs(project.getId(), launch.getId(), analyzerConfig);
 		}
+		eventPublisher.publishEvent(new LaunchEvent(launch.getId(), AUTO_ANALYSIS_FINISHED_KEY));
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java:74]: Can it be a separate class for this type of event?

**Review by pbortnik — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java]: replaced

File: src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java]: replaced

File: src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java [hunk 3]
```diff
@@ -39,22 +41,26 @@
 @Service
 public class LaunchAutoAnalysisSubscriber implements LaunchFinishedEventSubscriber {
 
+	public static final String AUTO_ANALYSIS_FINISHED_KEY = "autoAnalysisFinished";
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java:44]: Remove it

**Review by pbortnik — APPROVED** [review]

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java [hunk 4]
```diff
@@ -46,16 +50,20 @@
 @Transactional
 class StartLaunchHandlerImpl implements StartLaunchHandler {
 
+	public static final String START_KEY = "start";
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java:53]: Is it still needed?

**Review by pbortnik — APPROVED** [review]

File: src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/events/handler/subscriber/impl/LaunchAutoAnalysisSubscriber.java]: removed

File: src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java (hunk 4 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/launch/impl/StartLaunchHandlerImpl.java]: removed

**Pull request merged**